Notation.Plugin.AzureKeyVault/KeyVault/KeyVaultClient.cs

```diff
@@ -166,17 +165,25 @@ public async Task<byte[]> SignAsync(SignatureAlgorithm algorithm, byte[] payload
         /// </summary>
         public async Task<X509Certificate2> GetCertificateAsync()
         {
-            var cert = await _certificateClient.Value.GetCertificateVersionAsync(_name, _version);
-
-            // If the version is invalid, the cert will be fallback to 
-            // the latest. So if the version is not the same as the
-            // requested version, it means the version is invalid.
-            if (cert.Value.Properties.Version != _version)
+            KeyVaultCertificate cert;
+            if (string.IsNullOrEmpty(_version))
             {
-                throw new ValidationException($"Invalid certificate version. The user provides {_version} but the response contains {cert.Value.Properties.Version} as the version");
+                // If the version is not specified, get the latest version
+                cert = (await _certificateClient.Value.GetCertificateAsync(_name)).Value;
             }
-
-            return new X509Certificate2(cert.Value.Cer);
+            else
+            {
+                cert = (await _certificateClient.Value.GetCertificateVersionAsync(_name, _version)).Value;
+
+                // If the version is invalid, the cert will be fallback to 
+                // the latest. So if the version is not the same as the
+                // requested version, it means the version is invalid.
+                if (cert.Properties.Version != _version)
+                {
+                    throw new PluginException($"The version of the certificate retrieved from Azure Key Vault is different from the version specified in the request. The version specified in the request is {_version} but the version retrieved from Azure Key Vault is {cert.Properties.Version}");
```

Comment:
We probably want to provide an actionable next step for the user in this error message.

Reply:
Updated.